example/generated-src/cpp/sort_items.hpp

```diff
@@ -15,6 +16,33 @@ class SortItems {
 public:
     virtual ~SortItems() {}
 
+    /**
+     * Defines the name of the JNI C++ proxy class. Used to convert a
+     * C++ implemented object to a Java object when the type of the object being
+     * converted is unknown to the JniInterface (see djinni_support.hpp).
+     * 
+     * The proxy class name is only used for converting Djinni interfaces that
+     * are implemented in C++. Java implemented interfaces are converted differently.
+     * However, the C++ class of an interface implemented in Java must still have a
+     * jniProxyClassName method in order for Djinni's JniInterface::fromCpp method to compile.
+     * 
+     * @return The name of the class's associated JNI proxy class.
+     * 
+     * @see JniInterface in djinni_support.hpp
+     */
+    virtual const std::string jniProxyClassName() { return "com/dropbox/textsort/SortItems$CppProxy"; }
```

Comment:
The jniProxyClassName and objcProxyClassName are now generated for each interface to workaround type slicing when creating the Java and the Objective-C proxies. I would love to come up with a  better approach to this, but I think it will require a large change to Djinni's proxy caching. This was a "good enough" solution for our needs.